hdc/algo/dekad.py

```diff
@@ -79,6 +79,14 @@ def __eq__(self, other: object) -> bool:
             return self._dkd == other._dkd
         return False
 
+    def __lt__(self, other: object) -> bool:
+        """Check for less than inequality with other Dekad or a string."""
+        if isinstance(other, str):
+            return str(self) < other
+        if isinstance(other, Dekad):
+            return self._dkd < other._dkd
+        return False
```

Comment:
In fact we should just do this:

```suggestion
        return self._dkd < Dekad(other)._dkd
```

This will support all the "dekad-like" inputs and will raise ValueError when input is not "dekad-like". We then should also implement all the other ones in a similar fachion

- `__gt__`
- `__lt__`
- `__ge__`
- `__le__`

Reply:
Also tests!

Please add here: https://github.com/WFP-VAM/hdc-algo/blob/main/tests/test_dekad.py

Reply:
Although `date` and `datetime` inputs might need to be treated specially, I have not experimented with that.

Reply:
Hi @Kirill888, thanks for the suggestions. I have implemented them all. We can't implement your suggestion as is at it clashes with mypy checks, but I have implemented a variant of it.

Reply:
this looks good, for mypy there is always `# type: ignore` comment one can use. And yeah, `return NotImplemented` is proper, and not `raise ValueError()` I was suggesting, my bad.